File: Ahorn/entities/misc/PoisonGas.jl
module CommunalHelperPoisonGas

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/PoisonGas" SusanPoison(x::Integer, y::Integer, spritePath::String="objects/CommunalHelper/poisonGas/gas", radius::Integer=48)


const placements = Ahorn.PlacementDict(
   "Poison Gas (Communal Helper)" => Ahorn.EntityPlacement(
	  SusanPoison,
	  "point"
   )
)



function Ahorn.selection(entity::SusanPoison)
    x, y = Ahorn.position(entity)
    r = get(entity.data, "radius", 48)
    return Ahorn.Rectangle(x - r, y - r, r*2, r*2)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::SusanPoison, room::Maple.Room) = Ahorn.drawSprite(ctx, string(get(entity.data, "spritePath", "objects/CommunalHelper/poisonGas/gas"), "00"), 0, 0; sx=get(entity.data, "radius", 48)/24.0, sy=get(entity.data, "radius", 48)/24.0)

end

File: Loenn/entities/misc/poison_gas.lua
local utils = require('utils')
local drawableSprite = require('structs.drawable_sprite')

local poisonGas = {
    name = "CommunalHelper/PoisonGas",
    depth = -101
}
poisonGas.placements = {
    name = "main",
    data = {
        spritePath = "objects/CommunalHelper/poisonGas/gas00",
        radius = 48
    }
}
poisonGas.fieldInformation = {
    spritePath = {fieldType=path, allowFolders = false, allowFiles = true},
}

function poisonGas.selection(room, entity)
    return utils.rectangle(entity.x - entity.radius, entity.y - entity.radius, entity.radius*2, entity.radius*2)
end

function poisonGas.sprite(room, entity) 
    local sprite = drawableSprite.fromTexture(entity.spritePath, entity)
    sprite:setScale(entity.radius / 24, entity.radius / 24)
    return sprite
end

return poisonGas
